modules/billing/src/test/java/org/eclipse/xpanse/modules/billing/providers/HuaweiBillingServiceTest.java

```diff
@@ -4,70 +4,90 @@
 import org.eclipse.xpanse.modules.models.enums.DeployResourceKind;
+import org.eclipse.xpanse.modules.models.service.BillingDataResponse;
 import org.junit.jupiter.api.Assertions;
-import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
-import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.util.CollectionUtils;
 
 public class HuaweiBillingServiceTest {
 
     private final ObjectMapper mapper = new ObjectMapper();
 
-    @Autowired
-    HuaweiBillingService huaweiBillingService;
-
     @Test
     void onDemandBilling_null() throws IOException {
         DeployServiceEntity service = mapper.readValue(
                 new File("target/test-classes/service-test.json"),
                 DeployServiceEntity.class);
 
-//        List<BillingDataResponse> billings = huaweiBillingService.onDemandBilling(service, true);
-
+        HuaweiBillingService huaweiBillingService = new HuaweiBillingService();
         Assertions.assertThrows(NullPointerException.class,
                 () -> huaweiBillingService.onDemandBilling(null, true));
 
-//        Assertions.assertFalse(CollectionUtils.isEmpty(billings));
-
     }
 
-    @Disabled
     @Test
     void onDemandBilling() throws IOException {
         DeployServiceEntity service = mapper.readValue(
                 new File("target/test-classes/service-test-1.json"),
                 DeployServiceEntity.class);
-
-        for (DeployResourceEntity deployResourceEntity : service.getDeployResourceList()) {
-            if (DeployResourceKind.VM.equals(deployResourceEntity.getKind())) {
-                deployResourceEntity.setResourceId("8**");
-                deployResourceEntity.setName("8**");
-                Map<String, String> map = new HashMap<>();
-                map.put("", "");
-                deployResourceEntity.setProperty(map);
-            }
-            if (DeployResourceKind.DISK.equals(deployResourceEntity.getKind())) {
-                deployResourceEntity.setResourceId("8**");
-                deployResourceEntity.setName("8**");
-                Map<String, String> map = new HashMap<>();
-                map.put("", "");
-                deployResourceEntity.setProperty(map);
-            }
-
+        List<DeployResourceEntity> deployResourceEntityList = new ArrayList<>();
+        DeployResourceEntity deployResourceEntity1 = new DeployResourceEntity();
+        deployResourceEntity1.setId(UUID.fromString("e97f9443-3850-40be-812b-c823fc652979"));
+        deployResourceEntity1.setKind(DeployResourceKind.VM);
+        deployResourceEntity1.setResourceId("d681840d-4f91-4148-b73d-ff2bc66786f5");
+        deployResourceEntity1.setName("basic-wmm");
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+        try {
+            deployResourceEntity1.setCreateTime(sdf.parse("2023-03-29 10:16:37.543"));
+        } catch (ParseException e) {
+            e.printStackTrace();
```

Comment:
We should use log.error() to replace e.printStackTrace();

Reply:
ok